Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,257 @@
+
+```swift
+let x = /0; let y = 1/
+```
+
+Should this be considered to be two `let` bindings, with each initialization expression using prefix and postfix `/` operators, or is it a single regex literal?
+
+This also extends more generally to prefix and postfix operators containing the `/` character, e.g:
+
+```swift
+let x = </<0; let y = 1</<
+```
+Is this a regex literal `/<0; let y = 1</` with a prefix and postfix `<` operator applied, or two `let` bindings each using prefix and postfix `</<` operators?
+
+There are no easy ways of resolving these ambiguities, therefore a regex literal parsed with `/` delimiters will likely need to be introduced under a new language version mode, along with a deprecation of prefix and postfix `/` operators. Some prefix and postfix operators containing `/` may be disambiguated with parenthesis, but we may have to figure out a way to refer to the operator explicitly or deprecate prefix and postfix (but not infix) operators containing `/`.
+
+#### On the existing division operator `/`
+
+The existing division operator `/` has less concerns than the above cases, however it raises some cases that currently parse as a sequence of binary operations, whereas the user might be expecting a regex literal.
+
+For example:
+
+```swift
+extension Int {
+  static func foo() -> Int { 0 }
+}
+
+let x = 0
+/ 1 / .foo()
+```
+
+Today, this is parsed as a single binary operator chain `0 / 1 / .foo()`, with `.foo()` becoming an argument to the `/` operator. This is because while Swift does have some parser behavior that is affected by newlines, generally newlines are treated as whitespace, and expressions therefore may span multiple lines. However the user may well be expecting the second line to be parsed as a regex literal.
+
+This is also potentially an issue for result builders, for example:
+
+```swift
+SomeBuilder {
+  x
+  / y /
+  z
+}
+```
+
+Today this is parsed as `SomeBuilder { x / y / z }`, however it's likely the user was expecting this to become a result builder with 3 elements, the second of which being a regex literal.
+
+There is currently no source compatibility impact as both cases will continue to parse as binary operations. The user may insert a `;` on the prior line to get the desired regex literal parsing. However this may not be sufficient we may need to change parsing rules (under a version check) to favor parsing regex literals in these cases. We'd like to discuss this further with the community.
+
+It's worth noting that this is similar to an ambiguity that already exists today with trailing closures, for example:
+
+```swift
+SomeBuilder {
+  SomeType()
+  { print("hello") }
+  AnotherType()
+}
+```
+
+`{ print("hello") }` will be parsed as a trailing closure to `SomeType()` rather than as a separate element to the result builder.
```

Comment:
There's an even better analogy here to leading-dot syntax, which therefore isn't supported at the top level of a statement in a result builder.